### PR TITLE
[Notifications P2] Comment Moderation View

### PR DESCRIPTION
**Pull request by alpavanoglu**
Refs #https://github.com/Automattic/wordpress-mobile/issues/33

## Description
This PR adds the UI for comment moderation options sheet.

## Testing Steps
Go to `CommentModerationOptionsView` file in Xcode and enable Previews.
OR
Present the view in the app.

Compare against the designs here: yWt5gg3nWORhu079Qfv3mS-fi-1453_5937

## Screenshot Reference
| Light Mode  | Dark Mode |
| ------------- | ------------- |
|  Screenshot 2024-04-29 at 12 13 13 | Screenshot 2024-04-29 at 12 18 29 |

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dvdchr** [comment]
Hi @alpavanoglu 👋 , I'm bumping this PR's milestone to `24.9` since I'm starting code freeze. Feel free to re-target this to the release branch if this is a blocker or intended for `24.8`.

File: WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationOptionsView.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            .foregroundStyle(Color.DS.Foreground.primary)
+    }
+
+    private var optionsVStack: some View {
+        VStack(spacing: .DS.Padding.medium) {
+            ForEach(items, id: \.title) { item in
+                optionHStack(item: item)
+            }
+        }
+    }
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationOptionsView.swift:40]: @alpavanoglu I haven't tested the PR yet, but I don't see a button here to select an option? I think it makes sense for that logic to be part of this PR. What do you think?

Comment (alpavanoglu) [reply WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationOptionsView.swift:40]: Resolved in https://github.com/wordpress-mobile/WordPress-iOS/pull/23090/commits/9ccbdcee0e9fc1630a63a2a1ed9dcd16da4af63f

Wrapped the HStack in a Button

**Review by salimbraksa — APPROVED** [review]
LGTM! 🚀

**Pull request merged**